modules/caddyhttp/rewrite/caddyfile.go

```diff
@@ -110,3 +114,53 @@ func parseCaddyfileURI(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, err
 	}
 	return rewr, nil
 }
+
+func parseCaddyfileHandlePath(h httpcaddyfile.Helper) ([]httpcaddyfile.ConfigValue, error) {
+	if !h.Next() {
+		return nil, h.ArgErr()
+	}
+	if !h.NextArg() {
+		return nil, h.ArgErr()
+	}
+
+	// read the prefix to strip
+	path := h.Val()
+	if !strings.HasPrefix(path, "/") {
+		path = "/" + path
+	}
+
+	// the ParseSegmentAsSubroute function expects the cursor
+	// to be at the token just before the block opening,
+	// so we need to rewind because we already read past it
+	h.Reset()
+	h.Next()
+
+	// parse the block contents as a subroute handler
+	handler, err := httpcaddyfile.ParseSegmentAsSubroute(h)
+	if err != nil {
+		return nil, err
+	}
+	subroute, ok := handler.(*caddyhttp.Subroute)
+	if !ok {
+		return nil, h.Errf("segment was not parsed as a subroute")
+	}
+
+	// make a matcher on the path and everything below it
+	pathMatcher := caddy.ModuleMap{
+		"path": h.JSON(caddyhttp.MatchPath{path + "*"}),
+	}
```

Comment:
I figure we could validate the path argument a bit more - could make sure the user didn't suffix it with `*` themselves.

Reply:
Hm, yeah, this is an interesting case... I think we shouldn't change their input _at all_, though, so things will stay consistent. i.e. they should use `/foo/*` if they want `/foo/*` (but we should obviously strip the `*` when trimming the path -- but I think the path matcher should be consistent to the user.

If the matcher token doesn't start with `/`, then we should return an error. This directive only works with path matchers, so the user should enter one, even if we do treat it a little special... it should be consistent.